### PR TITLE
(MODULES-2520) Validate readonly params by default

### PR DESCRIPTION
**Pull request by ferventcoder**
If the value of the readonly parameter is not what we expect it
to be, fail validation for the parameter. Unfortunately using
properties and overriding property.retrieve did not work as intended.

Fallback to a method that does work appropriately. Validate the readonly
parameter is the same as the default value.

### CODE REVIEW

File: build/dsc/templates/dsc_type.rb.erb [hunk 1]
```diff
@@ -74,28 +74,24 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
...
+  newparam(:dscmeta_resource_friendly_name) do
     desc "A read-only value that is the DSC Resource Friendly Name ('<%= resource.friendlyname %>')."
 
-    def retrieve
-      '<%= resource.friendlyname %>'
-    end
-
     validate do |value|
-      fail 'dscmeta_resource_friendly_name is read-only'
+      fail 'dscmeta_resource_friendly_name is read-only' if value != '<%= resource.friendlyname %>'
```
Comment (Iristyle) [inline build/dsc/templates/dsc_type.rb.erb]: I think you can remove the `if` ... lets make it simpler and just fail if a user tries to set it at all (which is the method we use in AWS module - i.e. https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet/type/ec2_instance.rb)

Comment (ferventcoder) [inline build/dsc/templates/dsc_type.rb.erb]: That is exactly what we were trying to do before.

Comment (ferventcoder) [inline build/dsc/templates/dsc_type.rb.erb]: The problem is defaultto also counts as the user setting it.

Comment (Iristyle) [inline build/dsc/templates/dsc_type.rb.erb]: Gotcha, going to try to some things out...

Comment (Iristyle) [inline build/dsc/templates/dsc_type.rb.erb]: OK - so I think we can actually simplify this whole thing quite a bit, but departing from prior art.  Let me PR your PR to show you what I mean.

**Comment by Iristyle** [comment]
Superceded by #149

**Pull request closed without merging**